docs/content/examples/map-editor.tsx

```diff
@@ -14,17 +23,82 @@ const viewer = {
   username: 'demo-maxima',
 } as const;
 
+const toMapObject = (data: string | null): MapObject | null => {
+  if (!data) {
+    return null;
+  }
+
+  try {
+    return JSON.parse(data);
+  } catch {
+    return null;
+  }
+};
+
 export default function MapEditorExample() {
+  const location = useLocation();
+  const params = new URLSearchParams(location.search);
+
+  const [mapObject, setMapObject] = useState<MapObject | null>(
+    // Load map object from url if present
+    toMapObject(params.get('map')),
+  );
```

Comment:
Let's get rid of the comment, it's kinda obvious that that is what it does.

Can you also do `() => toMapObject(params.get('map'))` so it only runs on the first render?